rest/taskana-rest-spring/src/test/java/pro/taskana/rest/WorkbasketDefinitionControllerIntTest.java

```diff
@@ -144,27 +194,36 @@ void testFailOnImportDuplicates() throws IOException {
 
   @Test
   void testNoErrorWhenImportWithSameIdButDifferentKeyAndDomain() throws IOException {
-    ResponseEntity<List<WorkbasketDefinitionResource>> response =
-        template.exchange(
-            restHelper.toUrl(Mapping.URL_WORKBASKETDEFIITIONS) + "?domain=DOMAIN_A",
-            HttpMethod.GET,
-            restHelper.defaultRequest(),
-            new ParameterizedTypeReference<List<WorkbasketDefinitionResource>>() {});
-
-    List<String> list = new ArrayList<>();
-    ObjectMapper objMapper = new ObjectMapper();
-    WorkbasketDefinitionResource wbDef = response.getBody().get(0);
-    list.add(objMapper.writeValueAsString(wbDef));
-    wbDef.getWorkbasket().setKey("new Key for this WB");
-    list.add(objMapper.writeValueAsString(wbDef));
-    ResponseEntity<Void> responseImport = importRequest(list);
-    assertEquals(HttpStatus.NO_CONTENT, responseImport.getStatusCode());
+    List<WorkbasketDefinitionResource> wbList = getExportForDomain("DOMAIN_A").getBody();
+
+    String wbAsItIs = objMapper.writeValueAsString(wbList.get(0));
+    WorkbasketDefinitionResource differentLogicalId = wbList.get(0);
+    differentLogicalId.getWorkbasket().setKey("new Key for this WB");
+
+    // breaks the logic - should we really allow this case?
```

Comment:
Should I also fix this test?

When the same Id is given twice to different workbaskets then a link in the distribution targets to this id is ambigious.
(for example: {wb1: ID1, key1, domain1}, {wb2: ID1, key2, domain1}, {wb3: IDx,..., distributionTargets:[ID1]} - which workbasket is the distribution target?)

At the moment the distribution target would only be linked to the last workbasket which uses the id. 
This cannot be the expected behaviour.
We could link the distribution target to all workbaskets with the same Id.

In my opinion it's less confusing when we just deny this case.

Reply:
ID and key/domain are both unique identifiers for one workbasket. The case you decribe is invalid. Maybe this was a misunderstanding. Do you have the time to fix this?